bundles/com.espressif.idf.core/src/com/espressif/idf/core/toolchain/IDFTargets.java

```diff
@@ -0,0 +1,182 @@
+/*******************************************************************************
+ * Copyright 2025 Espressif Systems (Shanghai) PTE LTD. All rights reserved.
+ * Use is subject to license terms.
+ *******************************************************************************/
+
+package com.espressif.idf.core.toolchain;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.stream.Collectors;
+
+import com.espressif.idf.core.toolchain.targets.Architecture;
+import com.espressif.idf.core.toolchain.targets.Target;
+import com.espressif.idf.core.toolchain.targets.TargetSpec;
+import com.espressif.idf.core.toolchain.targets.internal.DynamicTarget;
+
+/**
+ * Class to hold ESP-IDF target information including preview status
+ * 
+ * @author Kondal Kolipaka <kondal.kolipaka@espressif.com>, Ali Azam Rana <ali.azamrana@espresif.com>
+ *
+ */
+public class IDFTargets
+{
+
+	private final List<IDFTarget> supportedTargets = new ArrayList<>();
+	private final List<IDFTarget> previewTargets = new ArrayList<>();
+
+	// ---------- Adders ----------
+
+	/** Strict for supported: unknown => throw (fail-fast). */
+	public void addSupportedTarget(String targetName)
+	{
+		Target known = Target.tryParse(targetName);
+		if (known == null)
+		{
+			throw new IllegalArgumentException("Unknown supported target: " + targetName); //$NON-NLS-1$
```

Comment:
I don’t think this is the right approach. From my understanding, the goal of this PR is to avoid adding new targets directly in the code and instead rely fully on IDF, under the assumption that new targets will use the RISC-V architecture.